kie-server-parent/kie-server-tests/kie-server-integ-tests-dmn/src/test/java/org/kie/server/integrationtests/dmn/modelspecific/ModelSpecificIntegrationTest.java

```diff
@@ -0,0 +1,309 @@
+ *  - show test strategy applied to previously existing endpoints (non model-specific)
+ *  - basic consistency test per each endpoint
+ *  - demonstrate error headers for non dmnresult endpoint
+ *  - demonstrate decision service coercion as specified in requirements, for non dmn result endpoint
+ *  - validate OAS produced for the containerId
+ *  - demonstrate model-specific endpoints are consistent with InputData definitions, such as JSON date/time.
+ */
+public class ModelSpecificIntegrationTest extends KieServerBaseIntegrationTest {
+
+    private static final String X_KOGITO_DECISION_MESSAGES = "X-Kogito-decision-messages";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ModelSpecificIntegrationTest.class);
+
+    private static final ReleaseId kjar1 = new ReleaseId("org.kie.server.testing", "ms-endpoints", "1.0.0.Final");
+
+    private static final String CONTAINER_ID = "ms-endpoints";
+    private static final String MODEL_NAME = "input-data-string";
+    private static final String MODEL_NAMESPACE = "https://github.com/kiegroup/kie-dmn/input-data-string";
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private CloseableHttpClient httpClient;
+
+    @BeforeClass
+    public static void deployArtifacts() {
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/ms-endpoints");
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    }
+    
+    @Before
+    public void cleanContainers() {
+        disposeAllContainers();
+        createContainer(CONTAINER_ID, kjar1);
+
+        // the new model-specific endpoint do not rely on the Kie-Server-Client Java API, hence using standard HttpClient
+        int timeout = 3;
+        RequestConfig config = RequestConfig.custom().setConnectTimeout(timeout * 1000).setConnectionRequestTimeout(timeout * 1000).setSocketTimeout(timeout * 1000).build();
+        httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+    }
+
+    @After
+    public void disposeHttpClient() throws Exception {
+        httpClient.close();
+    }
+
+    @Override
+    protected KieServicesClient createDefaultClient() throws Exception {
+        return null; // intentional, as the new model-specific endpoint do not rely on the Kie-Server-Client Java API.
+    }
+
+    @Override
+    protected void addExtraCustomClasses(Map<String, Class<?>> extraClasses) throws Exception {
+        // no extra classes.
+    }
+
+    private String baseContainerPath() {
+        return TestConfig.getKieServerHttpUrl() + "/containers/" + CONTAINER_ID;
+    }
+
+    private HttpResponse makePOST(String pathInContainer, String payload) throws Exception {
+        HttpPost httpPost = new HttpPost(baseContainerPath() + pathInContainer);
+        httpPost.setEntity(new StringEntity(payload));
+        httpPost.setHeader("Accept", "application/json");
```

Comment:
Is XML supported too? If so the test should be parametrized to check both marshallings.

Reply:
No, these new endpoints are only for JSON, per Requirements.